library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2184,8 +2210,16 @@ impl<T, A: Allocator> VecDeque<T, A> {
                 self.copy_nonoverlapping(0, tail + head_len, tail_len);
                 // ...ABCDEFGH.
             }
-
-            self.head = tail;
+            // according to benchmarks, it's faster to do this than to
+            // always fall back to `slice::rotate` if we want to left align.
```

Comment:
This refers to the else branch? If so the comment should only be there, not on the whole if-else.
And it can be phrased better, since as it is currently organized it could never reach the rotate-case from this point.

Reply:
In my mind the comment referred to the whole `if-else`, as in 'move tail, copy everything' is preferable to 'move head, rotate'. I'm not sure where you want me to move the comment to better indicate this.